pkg/api/api.go

```diff
@@ -160,29 +160,29 @@ func (a *API) RegisterAPI(cfg interface{}) {
 }
 
 // RegisterDistributor registers the endpoints associated with the distributor.
-func (a *API) RegisterDistributor(d *distributor.Distributor, pushConfig distributor.Config) {
-	a.RegisterRoute("/api/v1/push", push.Handler(pushConfig, d.Push), true)
+func (a *API) RegisterDistributor(d *distributor.Distributor, pushConfig distributor.Config, serverConfig server.Config) {
```

Comment:
**nitpick**: Forcing the serverconfig into each of these initialization functions seems like overkill for what we are trying to do.

**suggestion**: Instead of passing the serverConfig as a param to each of these functions. Would it be possible to create a single `middleware.SourceIPs` in the `initAPI` function and just have it as a set field in the API struct that is reused and passed directly to `push.Handler`?

Reply:
The SourceIpExtractor is now created (if needed) in the API struct and used in push.Handler. This change also exposed a bug. The SourceIpExtractor was created always even if the relevant config setting was false. This has been fixed now.